components/search_results/index.jsx

```diff
@@ -27,25 +27,38 @@ function makeMapStateToProps() {
     let isFlaggedByPostId;
 
     return function mapStateToProps(state) {
+        const config = getConfig(state);
+
+        const dataRetentionEnableMessageDeletion = config.DataRetentionEnableMessageDeletion === 'true';
+        const dataRetentionMessageRetentionDays = config.DataRetentionMessageRetentionDays;
+        const viewArchivedChannels = config.ExperimentalViewArchivedChannels === 'true';
+
         const newResults = getSearchResults(state);
 
         // Cache posts and channels
         if (newResults && newResults !== results) {
             results = newResults;
 
-            posts = results.filter((post) => Boolean(post));
-
             channels = new Map();
 
-            const channelIds = posts.map((post) => post.channel_id);
+            posts = [];
+            results.forEach((post) => {
+                if (!post) {
+                    return;
+                }
+
+                const channel = getChannel(state, post.channel_id);
```

Comment:
The old code would check the map first before triggering the `getChannel` selector -- should we do the same here?

Reply:
Maybe, we can move the archived channels checks after the map population to reuse the channel Map, and only call `getChannel` ones per channel.

Reply:
Fixed :)